### PR TITLE
POST Method to Send Follow Request

### PR DESCRIPTION
**Pull request by laylaylo**
### **How?**
POST API endpoint can be accessed via the "_follow_user/_" route and makes modifications to the Django database models. If the user is not signed in, the '_username_' and '_password_' of the follower user must be provided in the headers for authentication purposes. If the user is already signed in, the API will automatically follow the user, and the headers will be ignored. The '_followed_username_' must be provided in the data to specify the user that the follower wants to follow.

### **Testing?**

- The functionality has been thoroughly tested using _**unit tests**_.
- Additionally, the implementation has been tested via Postman after deployment on a local environment. A valid request example is available **_in our Postman workspace_**. Please note that you need to create the users that will be used in the request before calling this API.

### CODE REVIEW

File: practice_app/api/views.py [hunk 1]
```diff
@@ -421,3 +421,37 @@ def user_registration(request):
...
+        return JsonResponse({"status":"Username of followed should be provided."}, status = 404)
+    
+    if User.objects.filter(username=followed_username).exists():
+        followed_user = User.objects.get(username=followed_username)
+        if models.Follower.objects.filter(follower=user, followed=followed_user).exists():
+            return JsonResponse({"status":"You are already following this user."}, status=409)
+        elif models.FollowRequest.objects.filter(sender=user, receiver=followed_user).exists():
+            return JsonResponse({"status":"You have already sent a following request this user."}, status=409)
+        else:
+            models.FollowRequest.objects.create(sender=user, receiver=followed_user)
```
Comment (hakanaktas0) [inline practice_app/api/views.py:453]: status of the request should also be set.

**Review by hakanaktas0 — CHANGES_REQUESTED** [review]

File: practice_app/api/views.py (hunk 1 above)
Comment (laylaylo) [reply practice_app/api/views.py:453]: Good catch! I'll set it as `'pending'`.
Now I realized that I'm not allowing to send a follow request if it's been already sent to same user even it's got rejected. I believe this is a design choice and permitting sending second request after rejection is okay. What do you think?

Comment (hakanaktas0) [reply practice_app/api/views.py]: I think it is okay too.

**Review by hakanaktas0 — APPROVED** [review]
All looks good to merge.

**Pull request merged**